### PR TITLE
#2 Some Fixes 

### PR DESCRIPTION
**Pull request by Romazes**
#### Description

1. Updates the type of brokerage factory to TDAmeritradeBrokerageFactory
2. Fix multiple subscription/unsubscription to symbols
3. Setup Date.Time to `TimeZones.NewYork`
4. Keep userPrincipals in private variable (Now, we can do one request when login to system)
5. Fix: `getExchangeSymbol()` (`BF-B` -> `BF.B`)
6. Fix: WaitWebSocketResponse event. If we cannot get response from websocket then algorithm will stop.

#### How Has This Been Tested?

Run test in test project.
#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: QuantConnect.TDAmeritrade/TDAmeritradeSymbolMapper.cs [hunk 1]
```diff
@@ -61,9 +61,24 @@ public Symbol GetLeanSymbol(string brokerageSymbol, SecurityType securityType, s
...
+                brokerageSymbol.Replace('.', '/');
+            }
+            else if (brokerageSymbol.Contains('-'))
+            {
+                brokerageSymbol.Replace('-', 'p');
+            }
+            else if(brokerageSymbol.Contains('+'))
+            {
+                brokerageSymbol.Replace("+", "/WS");
+            }
```
Comment (Martin-Molinero) [inline QuantConnect.TDAmeritrade/TDAmeritradeSymbolMapper.cs:77]: No need to call `Contains` can directly call `Replace`?

File: QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.WebSocket.cs [hunk 2]
```diff
@@ -498,35 +499,19 @@ private void OnBestBidAskUpdated(object? sender, BestBidAskUpdatedEventArgs e)
...
-            {
-                if (!_symbolExchangeTimeZones.TryGetValue(symbol, out var exchangeTimeZone))
-                {
-                    // read the exchange time zone from market-hours-database
-                    exchangeTimeZone = MarketHoursDatabase.FromDataFolder().GetExchangeHours(symbol.ID.Market, symbol, symbol.SecurityType).TimeZone;
-                    _symbolExchangeTimeZones[symbol] = exchangeTimeZone;
-                }
-                return exchangeTimeZone;
-            }
-        }
```
Comment (Martin-Molinero) [inline QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.WebSocket.cs:528]: Should keep this method and not hard code `TimeZone.NewYork` usage in quotes and trades

File: QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.WebSocket.cs [hunk 3]
```diff
@@ -70,6 +71,13 @@ public partial class TDAmeritradeBrokerage
             { typeof(OrderRouteMessage), new XmlSerializer(typeof(OrderRouteMessage))}
         };
 
+        /// <summary>
+        /// We keep userPrincipals in first request when Login to system
+        /// Then we need some param to subscribe\unsubscribe in channels
+        /// Also, we decrease amount requests
+        /// </summary>
+        private UserPrincipalsModel userPrincipals;
```
Comment (Martin-Molinero) [inline QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.WebSocket.cs:79]: Not respecting code style -> `_userPrincipals`

File: QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.WebSocket.cs [hunk 4]
```diff
@@ -204,7 +212,7 @@ public void Login()
         {
```
Comment (Martin-Molinero) [inline QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.WebSocket.cs:212]: `Login()` method should be private, same `LogOut()`, please review the rest of the public methods 👍

File: QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.WebSocket.cs (hunk 3 above)
Comment (Romazes) [reply QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.WebSocket.cs]: I renamed to correct style type.

File: QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.WebSocket.cs (hunk 2 above)
Comment (Romazes) [reply QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.WebSocket.cs:528]: I came back this method. Also, I unhardcoded timeZone in quote and trade.

File: QuantConnect.TDAmeritrade/TDAmeritradeSymbolMapper.cs (hunk 1 above)
Comment (Romazes) [reply QuantConnect.TDAmeritrade/TDAmeritradeSymbolMapper.cs]: I changed to one line with helping 🙄 thanks

File: QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.WebSocket.cs (hunk 4 above)
Comment (Romazes) [reply QuantConnect.TDAmeritrade/TDAmeritradeBrokerage.WebSocket.cs:212]: I changed access modifier in these method and several in RestAPI file.

**Review by Martin-Molinero — APPROVED** [review]

**Pull request merged**